designdocs/provider_contexts.md

```diff
@@ -0,0 +1,354 @@
+    - BCP 47-compliant language codes (e.g., en-US, fr-FR).
+2. Updated `ProviderRegistryEntry` struct have the following properties:
+
+    ```rust
+        pub struct ProviderRegistryEntry<T::Config> {
+            pub default_name: BoundedVec<u8, T::MaxProviderNameSize>,
+            pub localized_names: BTreeMap<Vec<T::MaxLanguageCodeSize>, BoundedVec<u8, T::MaxProviderNameSize>>,
+            pub default_logo_250_100_png_bytes: BoundedVec<u8, T::MaxProviderLogo250X100Size>,
+            pub localized_logo_250_100_png_bytes: BTreeMap<Vec<T::MaxLanguageCodeSize>, BoundedVec<u8, T::MaxProviderLogo250X100Size>>,
+        }
+    ```
+
+3. New `ProviderToApplicationRegistryEntry` storage be initialized:
+
+    ```rust
+    // Alias for clarity
+    type ApplicationIdentifier<T: Config> = BoundedVec<u8, T::MaxProviderNameSize>
+
+    #[pallet::storage]
+    pub type ProviderToApplicationRegistryEntry<T: Config> = StorageDoubleMap<
+        _,
+        Twox64Concat,
+        ProviderId,
+        Twox64Concat,
+        ApplicationIdentifier<T>,
+        ProviderRegistryEntry<T>,
+        OptionQuery,
+    >;
+    ```
+
+4. `MaxProviderNameSize` be increased to `256`.
+5. `MaxProviderLogo250X100Size` be created and the limit set to `131_072` (128 KiB).
+6. Introduce hash-based logo approval mechanism:
+    Logos are not embedded during proposal submission. Instead, a blake2_256 hash of the logo image is included in the proposal.
+    Governance must explicitly approve these logo hashes as part of the provider registration process.
+
+    The approved hashes are recorded in a dedicated storage map:
+
+    ```rust
+            pub ApprovedLogoHashes: StorageMap<
+                _,
+                Blake2_128Concat,
+                [u8; 32], // blake2_256 hash of the logo
+                (),
+                OptionQuery
+            >;
+    ```
+
+    The `propose_to_be_provider` extrinsic will insert or update entries in this map.
+
+### **Changes and additions in extrinsics** <a id='extrinsics'></a>
+
+1. The `propose_to_be_provider` extrinsic will now accept an optional list of hashes for images/logos to be approved by governance.
+
+    ```rust
+        // Define a common payload type for provider registration
+        #[derive(Encode, Decode, Clone, PartialEq, Eq, RuntimeDebug)]
+        pub struct ProviderRegistrationPayload {
+            /// Default name for provider or application.
+            pub name: BoundedVec<u8, T::MaxProviderNameSize>,
+
+            /// A list of pre-approved logo image hashes (blake2_256).
+            pub logo_hashes: BoundedVec<[u8; 32], T::MaxApprovedLogoCount>,
+
+            /// Localized translations of the name, keyed by BCP 47 language code.
+            pub translations: BoundedBTreeMap<
+                BoundedVec<u8, T::MaxLanguageCodeSize>, 
+                BoundedVec<u8, T::MaxProviderNameSize>, 
+                T::MaxTranslations,
+            >,
+        }
```

Comment:
Proposed payload to register provider/application, this is minimal to get a bulk approved and update later